kotlin/create-symbol-styles-from-web-styles/src/main/java/com/esri/arcgisruntime/sample/createsymbolstylesfromwebstyles/MainActivity.kt

```diff
@@ -0,0 +1,239 @@
+                        scrollViewLayout.removeViewAt(symbolNames.indexOf(symbolName))
+                        // add the legend row at the correct index
+                        scrollViewLayout.addView(legendRow, symbolNames.indexOf(symbolName))
+                    }
+                } catch (e: Exception) {
+                    val error = "Error getting symbol: " + e.message
+                    Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+                    Log.e(TAG, error)
+                }
+            }
+        }
+
+        // add a map scale changed listener on the map view to control the symbol sizes at different scales
+        mapView.addMapScaleChangedListener {
+            featureLayer.isScaleSymbols = mapView.mapScale >= 80000
+        }
+    }
+
+    /**
+     * Returns a list of categories to be matched to a symbol name.
+     *
+     * @param symbolName the name of a symbol from a symbol style
+     * @return categories a list of categories matched to the provided symbol name
+     */
+    private fun mapSymbolNameToField(symbolName: String): List<String> {
+        return mutableListOf<String>().apply {
+            when (symbolName) {
+                "atm" -> add("Banking and Finance")
+                "beach" -> add("Beaches and Marinas")
+                "campground" -> add("Campgrounds")
+                "city-hall" -> addAll(
+                    listOf(
+                        "City Halls",
+                        "Government Offices"
+                    )
+                )
+                "hospital" -> addAll(
+                    listOf(
+                        "Hospitals and Medical Centers",
+                        "Health Screening and Testing",
+                        "Health Centers",
+                        "Mental Health Centers"
+                    )
+                )
+                "library" -> add("Libraries")
+                "park" -> add("Parks and Gardens")
+                "place-of-worship" -> add("Churches")
+                "police-station" -> add("Sheriff and Police Stations")
+                "post-office" -> addAll(
+                    listOf(
+                        "DHL Locations",
+                        "Federal Express Locations"
+                    )
+                )
+                "school" -> addAll(
+                    listOf(
+                        "Public High Schools",
+                        "Public Elementary Schools",
+                        "Private and Charter Schools"
+                    )
+                )
+                "trail" -> add("Trails")
+            }
+        }
+    }
+
+    /**
+     * Sets up UI behaviour. Closes expandable floating action button on touching the scene view.
+     * Moves floating action button on attribution view expanded. Expands floating action button on
+     * tap. Defines seek bar to control image overlay opacity. Populates and defines behaviour of the
+     * FPS (frames per second) spinner.
```

Comment:
i think the fps spinner ref is left over from a previous sample? Here it is responsible for populating and displaying the web style legend I think? Is it possible for this UI button to be blue to match the app title banner: or was there a reason we kept it green, can't remember!

Reply:
Oops! Yeah, copy paste error.

The green is one of the standard secondary colours we use in the other samples, so the button will match other samples. Agreed the colour scheme might need looking at in the future...